code/go/0chain.net/smartcontract/storagesc/blobber_reward_partition.go

```diff
@@ -37,14 +37,52 @@ func BlobberRewardKey(round int64) datastore.Key {
 	return sb.String()
 }
 
+func passedCallback(id string, data []byte, toPartition, _ int, sCtx c_state.StateContextI) error {
+	replace := &StorageNode{
+		ID: id,
+	}
+	if err := sCtx.GetTrieNode(replace.GetKey(ADDRESS), replace); err != nil {
+		return err
+	}
+	replace.LastRewardPartition.Index = toPartition
+	if _, err := sCtx.InsertTrieNode(replace.GetKey(ADDRESS), replace); err != nil {
+		return err
+	}
+	return nil
+}
+
 // getActivePassedBlobberRewardsPartitions gets blobbers passed challenge from last challenge period
 func getActivePassedBlobberRewardsPartitions(balances c_state.StateContextI, period int64) (*partitions.Partitions, error) {
 	name := BlobberRewardKey(GetPreviousRewardRound(balances.GetBlock().Round, period))
-	return partitions.CreateIfNotExists(balances, name, blobberRewardsPartitionSize)
+	blobbers, err := partitions.CreateIfNotExists(balances, name, blobberRewardsPartitionSize)
+	if err != nil {
+		return nil, err
+	}
+	blobbers.SetCallback(passedCallback)
```

Comment:
The reason for this is that we now want to remove blobbers from partitions if they have been killed. To do this, we have to keep the partition balanced, which means moving another blobber into the vacated place in the partition. For this all to work, we need to update the moved blobber with its new position in the partition so that it in turn, can be successfully killed if required.

This is part of the partition design, that when we remove an item from a partition, we call a callback function to be called when an item changes position in the partition. So I do not see adding this callback as a side effect, but part of the process of creating a partition.